catalog/plugin/catalog-plugin-pointofcontactupdate/src/test/java/org/codice/ddf/catalog/plugin/metacard/PointOfContactUpdatePluginTest.java

```diff
@@ -0,0 +1,183 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.metacard;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verifyZeroInteractions;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+
+public class PointOfContactUpdatePluginTest {
+
+    private static final String RESOURCE_ID1 = "resource-id1";
+
+    private static final String RESOURCE_ID2 = "resource-id2";
+
+    private static final String REGISTRY_ID = "registry-id";
+
+    private static final String ADMIN_EMAIL = "admin@localhost.local";
+
+    private static final String NEW_EMAIL = "new@email.com";
+
+    private PointOfContactUpdatePlugin pointOfContactUpdatePlugin =
+            new PointOfContactUpdatePlugin();
+
+    private List<Metacard> listOfUpdatedMetacards;
+
+    private UpdateRequestImpl updateRequestInput;
+
+    private Map<String, Metacard> existingMetacards;
+
+    @Before
+    public void setup() throws Exception {
+        listOfUpdatedMetacards = getListOfUpdatedMetacards();
+        updateRequestInput = new UpdateRequestImpl(new String[] {RESOURCE_ID1, RESOURCE_ID2,
+                REGISTRY_ID}, listOfUpdatedMetacards);
+        existingMetacards = getPreviousMetacardsWithPointOfContact();
+    }
+
+    @Test
+    public void processPreUpdateOnlyModifiesResourceMetacards() throws Exception {
+        UpdateRequest updateRequestOutput = pointOfContactUpdatePlugin.processPreUpdate(
+                updateRequestInput,
+                existingMetacards);
+
+        updateRequestOutput.getUpdates()
```

Comment:
❗️ Loops and conditionals are usually unit test smell. We should be able to have specific `assertThat` to verify that all the metacards we know about have the right POC information.